appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/resource/ResourceContentService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.kogito.bridge.client.resource;
+
+import elemental2.promise.Promise;
+
+/**
+ * Service to access resources in the project or workspace where the editor is open
+ *
+ */
+public interface ResourceContentService {
+
+    /**
+     * Returns a resource's content
+     * 
+     * @param uri
+     *  the resource URI relative to the workspace/project
+     * @return
+     * The resource content or null if the resource is not available
+     */
+    public Promise<String> get(String uri);
```

Comment:
As the service will return the raw String content for a resource, does that makes sense considering the charset as well on the arguments, in some way? Otherwise, the javacods at lesat should specify that charset is up to the given implementaton? Does that makese sense?

Reply:
I will double check with @tiagobento , but IMO it does not make a difference in the WEB ( ResourceContent is all client side) because Javascript uses UTF-16 for everything

Reply:
ok thanks